### ISSUE TITLE
IDE0200 is not reported in .NET 8 but works in .NET 7 for injected dependencies

### ISSUE THREAD

**Issue by arphox** [dotnet/roslyn#71300]
**Version Used**: I don't know how to check it.

I noticed that there is a difference between .NET 7 and .NET 8 in terms of what IDE0200 reports and I think it is a bug, since .NET 8 performs worse.

**Steps to Reproduce**:

`<TreatWarningsAsErrors>true</TreatWarningsAsErrors>`

editorconfig:
```
csharp_style_prefer_method_group_conversion = true:warning
dotnet_diagnostic.IDE0200.severity = warning
```

global.json – if I change version to `7.0.100`, it breaks the build. But for `8.0.100` it doesn't.
```
{
  "sdk": {
    "version": "8.0.100",
    "rollForward": "feature"
  }
}
```

code:
```cs
public class Repro
{
    private readonly MethodProvider _methodProvider;

    public Repro(MethodProvider methodProvider)
    {
        _methodProvider = methodProvider;
    }

    public void Main()
    {
        int[] numbers = { 1, 2, 3, 4, 5 };
        string[] asStrings = numbers.Select(x => _methodProvider.ToStr(x)).ToArray();
        Console.WriteLine(asStrings.Length);
    }
}

public class MethodProvider
{
    public string ToStr(int x)
    {
        return x.ToString();
    }
}
```

Notes:
1. I couldn't reproduce it with a simpler example where the called method would be in the same class.
2. The list of installed SDKs on my PC: `6.0.417`, `7.0.404`, `8.0.100`.

**Diagnostic Id**: IDE0200

**Expected Behavior**: analyzer reports

**Actual Behavior**: analyzer doesn't report

**Comment by CyrusNajmabadi** [comment]
I don't repro this at all.  This reports on all modern sdks for me.
